src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -393,46 +442,55 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
-                project?.AddFileToProjectAsync(_configFileName);
+                    await Shell.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                RunningDocumentTable rdt = new RunningDocumentTable(Shell.ServiceProvider.GlobalProvider);
-                string configFilePath = Path.GetFullPath(_configFileName);
+                    EnvDTE.Project project = VsHelpers.DTE.SelectedItems.Item(1)?.ProjectItem?.ContainingProject;
+                    project?.AddFileToProjectAsync(_configFileName);
 
-                IVsTextBuffer textBuffer = rdt.FindDocument(configFilePath) as IVsTextBuffer;
+                    RunningDocumentTable rdt = new RunningDocumentTable(Shell.ServiceProvider.GlobalProvider);
+                    string configFilePath = Path.GetFullPath(_configFileName);
 
-                _dispatcher.Invoke(() =>
-                {
-                    _closeDialog(true);
-                });
+                    IVsTextBuffer textBuffer = rdt.FindDocument(configFilePath) as IVsTextBuffer;
 
-                // The file isn't open. So we'll write to disk directly
-                if (textBuffer == null)
-                {
-                    manifest.AddLibrary(libraryInstallationState);
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
 
-                    await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                    // The file isn't open. So we'll write to disk directly
+                    if (textBuffer == null)
+                    {
+                        manifest.AddLibrary(libraryInstallationState);
+
+                        await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
 
-                    await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                        await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        // libman.json file is open, so we will write to the textBuffer.
+                        InsertIntoTextBuffer(textBuffer, libraryInstallationState);
+
+                        // Save manifest file so we can restore library files.
+                        rdt.SaveFileIfDirty(configFilePath);
+                    }
                 }
                 else
                 {
-                    // libman.json file is open, so we will write to the textBuffer.
-                    InsertIntoTextBuffer(textBuffer, libraryInstallationState);
-
-                    // Save manifest file so we can restore library files.
-                    rdt.SaveFileIfDirty(configFilePath);
+                    _isInstalling = false;
```

Comment:
>_isInstalling = false; [](start = 20, length = 22)

Is this necessary? I only ask as it wasn't happening before.

Reply:
Removed this in iteration 7 as this is not useful. The value will be reset as we open dialog for every install command.

My main intension was to set _isInstalling = true only when LibraryInstallationState is valid and close the dialog.

This is important as we want the dialog to remain open when message box is closed.

---
In reply to: [201160674](https://github.com/aspnet/LibraryManager/pull/293#discussion_r201160674) [](ancestors = 201160674)